markdown_it/extensions/amsmath/__init__.py

```diff
@@ -0,0 +1,103 @@
+"""An extension to capture ams math environments."""
+import re
+
+from markdown_it import MarkdownIt
+from markdown_it.rules_block import StateBlock
+from markdown_it.common.utils import escapeHtml
+
+# Taken from amsmath version 2.1
+# http://anorien.csc.warwick.ac.uk/mirrors/CTAN/macros/latex/required/amsmath/amsldoc.pdf
+ENVIRONMENTS = [
+    # 3.2 single equation with an automatically gen-erated number
+    "equation",
+    # 3.3 variation equation, used for equations that don’t fit on a single line
+    "multline",
+    # 3.5 a group of consecutive equations when there is no alignment desired among them
+    "gather",
+    # 3.6 Used for two or more equations when vertical alignment is desired
+    "align",
+    # allows the horizontal space between equationsto be explicitly specified.
+    "alignat",
+    # stretches the space betweenthe equation columns to the maximum possible width
+    "flalign",
+]
+# other "non-top-level" environments:
+
+# 3.4 the split environment is for single equations that are too long to fit on one line
+# and hence must be split into multiple lines,
+# it is intended for use only inside some other displayed equation structure,
+# usually an equation, align, or gather environment
+
+# 3.7 variants gathered, aligned,and alignedat are provided
+# whose total width is the actual width of the contents;
+# thus they can be used as a component in a containing expression
```

Comment:
In response to your question: http://anorien.csc.warwick.ac.uk/mirrors/CTAN/macros/latex/required/amsmath/amsldoc.pdf states that these should not be top-level environments. Have you found a use case where they are?